.github/workflows/test.yml

```
@@ -0,0 +1,128 @@
+name: Tests
+on:
+  push:
+    branches:
+      - develop
+  pull_request:
+    branches-ignore:
+      - master
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  tests:
+    name: Run ${{ matrix.mark}} tests (${{ matrix.python-version }} on ${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.7', '3.12']
+        os: [ubuntu-latest]
+        mark: [all]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build Backend Container
+        uses: docker/build-push-action@v5
+        with:
+          tags: backend:${{ github.sha }}
+          context: .
+          file: docker/Dockerfile.backend
+          push: false
+
+      - name: Start Backend Container
+        run: |
+          docker compose -f docker-compose.test.yml down -v --remove-orphans
+          docker compose -f docker-compose.test.yml up  --wait -d
+        env:
+          COMPOSE_PROJECT_NAME: ${{ github.run_id }}-backend
```

```suggestion
      - name: Build Backend Container
        uses: docker/build-push-action@v5
        with:
          tags: mtsrus/horizon-backend:${{ github.sha }}
          context: .
          file: docker/Dockerfile.backend
          push: false

      - name: Start Backend Container
        run: |
          docker compose -f docker-compose.test.yml down -v --remove-orphans
          docker compose -f docker-compose.test.yml up  --wait -d
        env:
          COMPOSE_PROJECT_NAME: ${{ github.run_id }}-backend
          TAG: ${{ github.sha }}
```

Ты собираешь образ, но называешь его не так, как указано в docker-compose.test.yml. Из-за этого docker-compose up собирает его еще раз, но уже без участия кэша (action сохраняет слои кэша для последующего использования, а docker-compose не умеет).